cetmix_tower_server/models/cx_tower_server_template.py

```diff
@@ -375,9 +374,53 @@ def _prepare_server_values(self, **kwargs):
             del values["id"]
             # update the values with additional arguments from kwargs
             values.update(kwargs)
+            # update server configs
+            values.update(server_config_values)
 
         return vals_list
 
+    def _parse_server_configs(self, config_values):
+        """
+        Prepares server configuration values.
+
+        Args:
+            config_values (dict): A dictionary containing server configuration values.
+                Keys and their expected values:
+                    - partner (res.partner, optional): The partner this server
+                      belongs to.
+                    - ipv4 (str, optional): IPv4 address. Defaults to None.
+                    - ipv6 (str, optional): IPv6 address. Must be provided if IPv4 is
+                      not specified. Defaults to None.
+                    - ssh_key (str, optional): Reference to an SSH private key record.
+                      Defaults to None.
+
+        Returns:
+            dict: A dictionary containing parsed server configuration values with the
+                following keys:
+                    - partner_id (int, optional): ID of the partner.
+                    - ssh_key_id (int, optional): ID of the associated SSH key.
+                    - ip_v4_address (str, optional): Parsed IPv4 address.
+                    - ip_v6_address (str, optional): Parsed IPv6 address.
+        """
+        values = {}
+
+        if config_values.get("partner"):
+            values["partner_id"] = config_values.pop("partner").id
+
+        if config_values.get("ssh_key"):
+            ssh_key_reference = config_values.pop("ssh_key")
+            ssh_key = self.env["cx.tower.key"].get_by_reference(ssh_key_reference)
+            if ssh_key:
```

Comment:
Just to improve performance:

```python
partner = config_values.pop("partner"):
if partner:
    values["partner_id"] = partner.id

ssh_key_reference = config_values.pop("ssh_key")
if ssh_key_reference:
    ssh_key = self.env["cx.tower.key"].get_by_reference(ssh_key_reference)
        if ssh_key:
            values["ssh_key_id"] = ssh_key.id
```
 Same with `ipv4` and `ipv6`

Reply:
we will get an error if `config_values` does not contain any key
![image](https://github.com/user-attachments/assets/14d2ccc7-2f3d-4335-8aa6-b83986284603)

Reply:
fixed